### PR TITLE
add cmocka testsuites for syscall test

### PR DESCRIPTION
**Pull request by vela-mib**
## Summary
add cmocka testsuites including mm_test, sched_test, pthread_test and time_test
## Impact
CI run
## Testing
local host

### CODE REVIEW

File: testing/testsuites/built
Comment (xiaoxiang781216) [inline testing/testsuites/built:1]: why add empty  file

File: testing/testsuites/Makefile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+include $(APPDIR)/Make.defs
+
+
+PRIORITY = $(CONFIG_TESTS_TESTSUITES_PRIORITY)
+STACKSIZE = $(CONFIG_TESTS_TESTSUITES_STACKSIZE)
+MODULE = $(CONFIG_TESTS_TESTSUITES)
+
+# treat all warning as error
+CFLAGS += -Werror
```
Comment (xiaoxiang781216) [inline testing/testsuites/Makefile:23]: remove

File: testing/testsuites/Makefile [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+include $(APPDIR)/Make.defs
+
+
+PRIORITY = $(CONFIG_TESTS_TESTSUITES_PRIORITY)
+STACKSIZE = $(CONFIG_TESTS_TESTSUITES_STACKSIZE)
+MODULE = $(CONFIG_TESTS_TESTSUITES)
+
+# treat all warning as error
+# CFLAGS += -Werror
```
Comment (xiaoxiang781216) [inline testing/testsuites/Makefile:23]: remove directly

Comment (vela-mib) [reply testing/testsuites/Makefile]: > remove directly

done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**